Add transform spec for performance measures

alightwing opened the pull request:
Previous transforms have been relatively simple. This one was definitely not. We have many ontology classes encapsulated in this sheet with a complex set of relationships that wasn't an exact match for the source data -- there are some conceptual differences and broken linkages. I am not sure this is usable as-is as none of these objects are visible or easily reachable from things that are.

I'll write up some notes tomorrow morning so that I haven't forgotten everything by the time we do the post-mortem next week (hopefully).

Other misc changes:

* There is an `as_date_or_text` property in `field` now -- this is to try and deal with a column containing mixed date formats where Excel has helpfully converted half of it to a full yyyy/mm/dd date (which is ingested as a Python datetime and needs bits chopping off to be sane) and the other half remains as the original text.
* Some other changes to stringifying that duplicate changes from the other PR, as discussed.
* A fix for specs falling over when trying to dump the model back out to JSON.

bz2 commented on sheet_to_triples/rdf.py:
```diff
@@ -152,6 +152,10 @@ def normalise_model(model, ns, verbose):
                 s=_n3(key[0], ns), p=_n3(key[1], ns), o=by_key[key]['obj']))
         by_key[key] = t
 
+    # set can't be written as JSON and we don't need this any more, so get rid
+    # of it
+    del model['non_uniques']
```
Yeah, this is a bit awkward, have put up alternative approach in this branch:
https://github.com/VisualMeaning/sheet-to-triples/pull/39

bz2 commented on sheet_to_triples/field.py:
```diff
@@ -59,7 +59,7 @@ def as_uc(self):
     @property
     def as_text(self):
-        return _must(self._value.strip())
+        return _must(str(self).strip())
```
Not completely sure about this, as it feels like we need a way of rejecting a column full of numbers or dates if we're expecting text, and the python stringification is sometimes just the wrong thing, eg for `None` from an empty cell.

alightwing replied:
So I kind of dealt with this problem with the `date_or_text` hack -- two implementations of the method, the strict one which falls over if the source data is the wrong format, and the flabby one which wraps a `try/except` around the strict one and does something else if it fails.  But then writing two methods for every cell value conversion doesn't exactly scale.

Ultimately I feel that this is not a problem that we can apply strict type requirements to -- or at least, not a problem solved by applying strict type requirements to. N consultants making spreadsheets are going to find N different ways of structuring the data, and then Excel jumps in to helpfully mangle it some more, and then you've got human error involved where somebody makes a typo or misformats a cell and the resulting bad value makes ingest of a 6,000 row spreadsheet fall over completely. Either the process for finding and fixing bad values in the source data needs to be much better than it currently is (i.e. practically non-existent) or we're going to have to be somewhat generous with how we interpret it during ingest.

alightwing replied:
Although an interim solution could be to restructure to reject bad values without the entire thing blowing up.

bz2 reviewed (approved):
Thanks! A few things to follow up on, and also your questions from the transform/ontology are worth picking up later too.

alightwing commented:
@bz2 a couple of changes before merging:

* removed string casting from `field.py` -- this will have to be hacked back in in some fashion if you want to actually run the transform as the source data is still inconsistent, but I agree that the default versioned properties should be strict on data types.
*  added a new column to the performance measures sheet which fixes the `DesiredResult` linking to match what's being exported from AI
